src/mpid/ch4/shm/xpmem/xpmem_avl.h

```diff
@@ -0,0 +1,564 @@
+    goto fn_exit;
+}
+
+
+/****************************************/
+/* AVL tree search related functions */
+/****************************************/
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_avl_tree_node_search
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX void *MPIDI_XPMEM_avl_tree_node_search(MPIDI_XPMEM_seg_t *node,
+                                                                uint64_t remote_low,
+                                                                uint64_t remote_high)
+{
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_XPMEM_AVL_TREE_NODE_SEARCH);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_XPMEM_AVL_TREE_NODE_SEARCH);
+    while (node) {
+        /* remote_low and remote_high must be between minimal and maximal addr */
+        if (MPIDI_XPMEM_AVL_TREE_RLOW_LOCATE_MIN_RIGHT(node, remote_low) &&
+            MPIDI_XPMEM_AVL_TREE_RHIGH_LOCATE_MAX_LEFT(node, remote_high)) {
+            /* Check seg of current node */
+            if (MPIDI_XPMEM_AVL_TREE_RLOW_LOCATE_LOW_LEFT(node, remote_low)) {
+                /* Can only find match on left child */
+                node = node->left;
+                continue;
+            } else {
+                if (MPIDI_XPMEM_AVL_TREE_RHIGH_LOCATE_HIGH_LEFT(node, remote_high)) {
+                    /* Match a cache */
+                    uint64_t offset = remote_low - node->low;
+                    return (void *) ((uint64_t) node->vaddr + offset);
+                } else {
+                    /* Can only find match on right child */
+                    node = node->right;
+                    continue;
+                }
+            }
+        } else
+            break;
+    }
+
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_XPMEM_AVL_TREE_NODE_SEARCH);
+    return NULL;
+  fn_fail:
+    goto fn_exit;
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_seg_avl_tree_search
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_seg_avl_tree_search(MPIDI_XPMEM_avl_tree_t * tree,
+                                                             MPIDI_XPMEM_seg_t * seg)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_XPMEM_SEG_AVL_TREE_SEARCH);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_XPMEM_SEG_AVL_TREE_SEARCH);
+    if (tree->root == NULL)
+        seg->vaddr = NULL;
+    else
+        seg->vaddr = MPIDI_XPMEM_avl_tree_node_search(tree->root, seg->low, seg->high);
+
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_XPMEM_SEG_AVL_TREE_SEARCH);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+
+/****************************************/
+/* AVL tree insertion related functions */
+/****************************************/
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_XPMEM_update_node_info
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_update_node_info(MPIDI_XPMEM_seg_t * node)
+{
```

Comment:
Looks like you name all internal routines with prefix `MPIDI_XPMEM_avl_tree_`, can you also add it for this function ?